storage/src/mmr/mem.rs

```diff
@@ -153,6 +153,30 @@ impl<H: CHasher> Mmr<H> {
         element_pos
     }
 
+    /// Pop the most recent leaf element out of the MMR (if it exists).
+    pub fn pop(&mut self) -> Result<u64, Error> {
```

Comment:
Can you provide an argument to specify the `new_size` rather than assuming `self.size()-1`?

Reply:
The implementation is popping off a single leaf and all its associated parents, rather than backing up to an arbitrary size point.  We could accept a parameter to specify how many leaves to pop off, but then the edge case handling gets a bit messy. We don't know exactly how many total nodes we'd be removing, so if we cross a pruning boundary in the middle of it, we'd probably want to restore all the nodes we'd previously removed to make it "atomic".

Reply:
As discussed, we'll move forward with this as is 👍